src/mkdocs_llmstxt/_internal/plugin.py

```diff
@@ -88,64 +101,117 @@ def on_files(self, files: Files, *, config: MkDocsConfig) -> Files | None:  # no
         Returns:
             Modified collection or none.
         """
-        for file in self.config.files:
-            file["inputs"] = self._expand_inputs(file["inputs"], page_uris=list(files.src_uris.keys()))
+        page_uris = list(files.src_uris)
+
+        for section_name, file_list in list(self.config.sections.items()):
+            self.config.sections[section_name] = self._expand_inputs(
+                file_list, page_uris=page_uris
+            )
+
         return files
 
     def on_page_content(self, html: str, *, page: Page, **kwargs: Any) -> str | None:  # noqa: ARG002
-        """Record pages contents.
+        """Convert page content into a Markdown file and save the result to be processed in the `on_post_build` hook.
 
         Hook for the [`on_page_content` event](https://www.mkdocs.org/user-guide/plugins/#on_page_content).
-        In this hook we simply record the HTML of the pages into a dictionary whose keys are the pages' URIs.
 
         Parameters:
             html: The rendered HTML.
             page: The page object.
         """
-        for file in self.config.files:
-            if page.file.src_uri in file["inputs"]:
-                _logger.debug(f"Adding page {page.file.src_uri} to page {file['output']}")
-                self.html_pages[file["output"]][page.file.src_uri] = html
+        for section_name, file_list in self.config.sections.items():
+            if page.file.src_uri in file_list:
+                path_md = Path(page.file.abs_dest_path).with_suffix(".md")
+                page_md = generate_page_markdown(
+                    html, self.config.autoclean, self.config.preprocess
+                )
+
+                md_url = Path(page.file.dest_uri).with_suffix(".md").as_posix()
+                if md_url in (".", "./"):
+                    md_url = ""
+                md_url = urljoin(
+                    # Guaranteed to exist as we require 'site_url' to be configured:
+                    cast(str, self.mkdocs_config.site_url),
+                    md_url,
+                )
+
+                self.md_pages[section_name].append(
+                    MDPageInfo(
+                        title=cast(
+                            str,
+                            page.title if page.title is not None else page.file.src_uri,
+                        ),
```

Comment:
nit: I generally prever type-ignore comments over casts, because they're not reported as "unused" for false-positives (contrary to type-ignore comments) when Mypy's abilities are improved.

No strong opinion though.

Reply:
Actually this was only erroring with pyright, because `title` is decorated with an untyped descriptor. Mypy seems to be fine.